### PR TITLE
ModifierEffects stored as instances

### PR DESCRIPTION
**Pull request by Hop311**
- Changed `ModifierEffect`s to be stored in `unique_ptr`s, so that pointers and references to them are stable without the need for locking. This allows us to add effects mid-load, even after having already parsed other modifiers.
- Added `PROPORTION_DECIMAL` as an effect format representing unscaled fractions, e.g. 0.01 or -0.2 (corresponding to +1% and -20% respectively). This used to be represented by `PERCENTAGE_DECIMAL`, which now instead represents fractions multiplied by 100 to be actual percentages, e.g. 15 or -40 (corresponding to +15% and -40% respectively).
- Removed hardcoded `max_***` and `min_build_***` building modifier effects, instead generating them for each building loaded into the `BuildingManager`.

- Fixed some TGC compatibility issues in country loading, one failure remains while I work out how best to address it (a TGC placeholder policy `no_position_set` is defined as a `social_policy` but used for other types in `countries/Zombie Wasteland.txt`).

- Changed the return type form of ‘Date‘ from a const ref to a copied value, since it's really just an ‘int64_t‘ under the hood.

### CODE REVIEW

File: src/openvic-simulation/Modifier.cpp
Comment (schombert) [inline src/openvic-simulation/Modifier.cpp:106]: it is a bit cleaner to use `std::make_unique` for this

File: src/openvic-simulation/Modifier.cpp
Comment (schombert) [inline src/openvic-simulation/Modifier.cpp:201]: If you don't want to hard-code the building max modifiers, you may not want to hard-code these either as they are derived from the names of the tech categories. Does any mod add a tech category? That I don't know.

File: src/openvic-simulation/Modifier.cpp
Comment (Hop311) [reply src/openvic-simulation/Modifier.cpp:106]: I couldn't because `ModifierEffect`'s constructor is `private`, accessible by `ModifierManager` because it's a friend. Thinking about it again, I should be able to make `std::make_unique` a friend instead. That will publicly expose `ModifierEffect` construction though, at which point I might as well just make the constructor public and not need a friend declaration.

File: src/openvic-simulation/Modifier.cpp
Comment (Hop311) [reply src/openvic-simulation/Modifier.cpp:201]: Thanks for letting me know, dunno about mods changing categories but since we're making it possible to change them then the changes should propagate all the way through.

File: src/openvic-simulation/Modifier.cpp
Comment (Spartan322) [reply src/openvic-simulation/Modifier.cpp:106]: Honestly making the constructor private makes little sense if the registry is locked, the member variables don't even have to be private (or const) necessarily if we disable the const functions of the registry, should however rename the non-const overload functions to something like `<name>_nc` or `<name>_mutable to clarify non-const/mutable return value.

**Review by BrickPi — APPROVED** [review]
Looks good generally, primarily reviewed history changes, probably want to wait for Spartan to check off before merge.

**Review by Spartan322 — CHANGES_REQUESTED** [review]
Could this be squashed a little bit at least, at minimum enough to remove the commits responsible for only fixing things that target this PR? (like especially if a commit was not building, it might as well be ignored, but also like fix and tweak commits should be squashed down to keep the PR's commits at least more relevant, making regression testing easier)

**Comment by Hop311** [comment]
> Could this be squashed a little bit at least, at minimum enough to remove the commits responsible for only fixing things that target this PR? (like especially if a commit was not building, its existence might as well be ignored, but also like fix and tweak commits should be squashed down to keep the PR's commits at least more relevant, making regression testing easier)

I agree most commits could do with being renamed, and as for squashing I'm happy to where it makes sense, did you have any particular commits in mind (going off the actual contents of the commits, rather than just their names)?

**Review by Spartan322 — APPROVED** [review]

**Pull request merged**
